### DISCUSSION TITLE
Cookies in a route handler

### DISCUSSION THREAD

**Discussion by adamchipperfield** [vercel/next.js#59125] [Help]
### Summary

Here's my use case and, possibly misguided, expectation:

I have a Route Handler at /cart/get. It looks for a cart ID cookie, fetches the cart and returns it as a JSON response.

If the cookie doesn't exist, I create a new cart and return that as a JSON response. **I also return a `Set-Cookie` header to set the new cookie.**

However, when I run that initial search for the cookie it _always_ comes out undefined. I can see the cookie in my browser, (from `Set-Cookie`) but it isn't found in the route handler with `cookies().get('cart_id')`.

Am I misunderstanding how cookies work here?

```js
export async function GET() {
  if (cookies().has('cart_id')) { // <--- issue: this always comes out false

    // cart cookie exists, fetch and return the cart

    return Response.json(
      await getCart(cookies().get('cart_id'))
    )
  }

  // create a new cart, return it and set the cookie

  const cart = await createCart()

  return Response.json(
    cart,
    {
      headers: {
        'Set-Cookie': `cart_id=${cart.id}`
      }
    }
  )
}
```

**Comment by ananducherakkal** [comment]
Were you able to find any solution? I am having the same issue.
